server/models/triggers.sql

```diff
@@ -114,9 +114,21 @@ FOR EACH ROW BEGIN
   ON DUPLICATE KEY UPDATE uuid = NEW.document_uuid;
 END$$
 
--- Stock Requisition Triggers
-CALL add_column_if_missing("stock_requisition", "project_id", "SMALLINT(5) UNSIGNED NOT NULL AFTER `user_id`");
+-- Shipment Triggers
+-- the shipment reference is incremented based on the shipment uuid.
+CREATE TRIGGER shipment_reference BEFORE INSERT ON shipment
+FOR EACH ROW
+  SET NEW.reference = (SELECT IF(NEW.reference, NEW.reference, IFNULL(MAX(sH.reference) + 1, 1)) FROM shipment sh WHERE sh.uuid <> NEW.uuid);$$
```

Comment:
TYPO!  Note the 'sH.reference' should be 'sh.reference'.  I had to fix this before I could run the tests.

Reply:
RESOLVED 👍🏽